text/0000-road-to-typescript.md

```diff
@@ -0,0 +1,160 @@
+
+## Motivation
+
+TypeScript is a key ingredient of contemporary front-end development. Per community surveys like the GitHub Octoverse, the Stack Overflow Developer report, and the State of JS survey, TypeScript has exploded in popularity over the last half decade, and is the rare tool which has continuously grown in both both usage and satisfaction over that time. ( While all of these surveys should be understood to be wildly unrepresentative of the broader front-end development community in various ways, they nonetheless provide some useful signal about the state of the ecosystem.)
+
+Additionally, the development of the language server protocol and the widespread adoption of the TypeScript language server has led to first-class support for TypeScript across many tools, from Vim to Visual Studio. This support also powers rich developer experiences for JavaScript-only developers, as it can provide smart autocomplete, inline documentation, and more even for codebases which are written solely in JavaScript. Thus, non-TypeScript and TypeScript users alike benefit when working with a project which supplies type definitions.
+
+Ember users have been working with TypeScript for years with community support led by the Typed Ember team, but there is one critical problem which cannot be solved without making TypeScript support official, and several key ways in which official support would substantially improve the experience of both TypeScript and JavaScript developers:
+
+- [ ] the SemVer problem
+    - DefinitelyTyped
+    - TS version support policy
+- [ ] other key benefits for official adoption
+    - coordination (e.g. editions aren't done till docs also goes for TS and types)
+    - [ ] blueprints?
+    - [ ] more?
+
+## Detailed design
+
+The primary goal of this RFC is to add TypeScript as an officially supported language. This involves four major elements:
+
+- Identifying the [constraints](#constraints) under which adoption can proceed
+- Making explicit the [non-goals](#non-goals) of this project
+- [Defining official support](#official-support): what it does and does not entail
+- Building a [roadmap](#roadmap) for completing the effort
+
+### Constraints
+
+The following are our hard constraints in supporting TypeScript with Ember:
+
+- We must not compromise our commitment to “stability without stagnation” and our strong Semantic Versioning guarantees. 
+- Using TypeScript should never be mandatory for anyone who uses Ember.
+- TypeScript support may never *degrade* the experience of JavaScript users, and wherever possible it should benefit both JavaScript and TypeScript developers.
+
+### Non-Goals
```

Comment:
- [ ] improve the definition of "non-goals": not included in initial road to TS
- [ ] include "not even changing the default" in second section

Reply:
54ac11d